diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -159,6 +160,14 @@ fun ASTNode.replaceWhiteSpaceText(beforeNode: ASTNode, text: String) {
 fun ASTNode.getFirstChildWithType(elementType: IElementType): ASTNode? =
         this.findChildByType(elementType)
 
+/**
+ * Checks if a function is anonymous
+ */
+fun ASTNode.isAnonymousFunction() {
+    require(this.elementType == FUN)
```

Comment:
can it be a part of checking of this function instead of assumption?

Reply:
I don't think it's needed here. We already everywhere have a check that this value is equal to FUN before calling this method

Reply:
so this function treats to node to be a `FUN`. it means that it's can be called only
`node.elementType == FUN && node.isAnonymousFunction()`, otherwise -- exception

Reply:
Yes, that's the point. We do not want the developer to try to call this function on a different type of node, In this case, we will not know what to return